pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -148,25 +153,6 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 		return nil, maxMonID, monMapping, errors.Wrap(err, "failed to get mon config")
 	}
 
-	// If an admin key was provided we don't need to load the other resources
-	// Some people might want to give the admin key
-	// The necessary users/keys/secrets will be created by Rook
-	// This is also done to allow backward compatibility
-	if clusterInfo.CephCred.Username == cephclient.AdminUsername && clusterInfo.CephCred.Secret != adminSecretNameKey {
-		return clusterInfo, maxMonID, monMapping, nil
-	}
-
-	// If the admin secret is "admin-secret", look for the deprecated secret that has the external creds
-	if clusterInfo.CephCred.Secret == adminSecretNameKey {
```

Comment:
The adminSecretNameKey and everything using it is really obsolete, so I've removed it and fixed the unit tests.

Reply:
thanks for fixing